File: JAVA/jzbus/src/test/java/net/zbus/reqrep/ReqRepWorker.java
package net.zbus.reqrep;

import net.zbus.Connection;
import net.zbus.ConnectionConfig;
import net.zbus.Worker;
import net.zbus.WorkerConfig;
import net.zbus.ZMsg;

public class ReqRepWorker {
 
	public static void main(String[] args) { 
		WorkerConfig workerCfg = new WorkerConfig();
		workerCfg.setService("helloworld");
		
		
		ConnectionConfig connCfg = new ConnectionConfig();  
		//connCfg.setVerbose(true);
		Connection conn = new Connection(connCfg);
		
		Worker worker = new Worker(conn, workerCfg);
		
		while(true){
			try{
				ZMsg msg = worker.recv();
				//msg.dump(); 
				worker.send(msg);
			}catch (Exception e) {
				e.printStackTrace();
				break;
			}
		}
		worker.destroy();
	}

}

File: trunk/MS.NET/clrzbus/examples/reqrep/ReqRepWorker.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using zbus;
namespace zbus
{
    class ReqRepWorker
    {
        public static void Main(string[] args)
        {
            ConnectionConfig connCfg = new ConnectionConfig();
            connCfg.Host = "127.0.0.1";
            connCfg.Port = 15555;
            Connection conn = new Connection(connCfg);

            WorkerConfig workerCfg = new WorkerConfig();
            workerCfg.Service = "helloworld";

            Worker worker = new Worker(conn, workerCfg);
            while (true)
            {
                try
                {
                    ZMsg msg = worker.Recv();
                    msg.Dump();
                    msg.PushBack("from C#");
                    worker.Send(msg);
                }
                catch (Exception e)
                {
                    Console.WriteLine(e);
                    break;
                }
            }
            conn.Destroy();
            worker.Destroy();
        } 
    }
}
